components/MailButton.tsx

```diff
@@ -19,7 +31,20 @@ const MailButton: VFC = () => {
   if (!data) {
     return <Loading />;
   }
-  return <Button onClick={sendTickets}>Send ut billetter</Button>;
+  return (
+    <Button
+      onClick={() =>
+        sendTickets(
+          data?.filter(
+            (attendee) =>
+              attendee.email_sent === false && attendee.active === true
+          )
+        )
+      }
```

Comment:
I agree with Peter here, I think the filtering should be done in the endpoint. I think the request shouldn't include any attendees at all and then we can rather fetch attendees with email_sent=false in the endpoint and send the mails one by one there instead of bulk (but not by calling the endpoint one by one linke this). But we should refactor this code anyways and use [https://nodemailer.com/smtp/pooled/](https://nodemailer.com/smtp/pooled/) as Ludvig pointed out on slack.

Reply:
Yeah, the issue is this that the server less functions at Vercel have a timeout of 60 seconds. Thus, we would easily time out when trying to process 600 attendees. So I think we have to call our Next API multiple times anyway.
[https://vercel.com/docs/concepts/limits/overview](https://vercel.com/docs/concepts/limits/overview)

Reply:
So the ultimate solution is probably to have some kind of timer in our send_emails endpoint:
So ish as follows:
When the send_email endpoint is called, the API fetches all attendees that have not received an email yet.
The Api starts a timer of maybe 45 seconds, and starts to send requests to the smtp-relay, and updating email_sent in our database.
When is has been 45 seconds (15 seconds before the server less function shuts down),
the API returns for instance
successfully sent email
rejected emails
remaining emails if any

Reply:
Peter fixed my solution. Now we happy happy.